public/pages/HistoricalDetectorDetail/containers/HistoricalDetectorDetail/HistoricalDetectorDetail.tsx

```diff
@@ -0,0 +1,410 @@
+  useEffect(() => {
+    if (
+      detector?.curState === DETECTOR_STATE.RUNNING ||
+      detector?.curState === DETECTOR_STATE.INIT
+    ) {
+      const intervalId = setInterval(
+        fetchDetector,
+        HISTORICAL_DETECTOR_RESULT_REFRESH_RATE
+      );
+      return () => {
+        clearInterval(intervalId);
+      };
+    }
+  }, [detector]);
+
+  const handleDeleteAction = () => {
+    setHistoricalDetectorDetailModalState({
+      ...historicalDetectorDetailModalState,
+      isOpen: true,
+      action: HISTORICAL_DETECTOR_ACTION.DELETE,
+    });
+  };
+
+  const handleEditAction = () => {
+    detector?.curState === DETECTOR_STATE.RUNNING ||
+    detector?.curState === DETECTOR_STATE.INIT
+      ? setHistoricalDetectorDetailModalState({
+          ...historicalDetectorDetailModalState,
+          isOpen: true,
+          action: HISTORICAL_DETECTOR_ACTION.EDIT,
+        })
+      : props.history.push(`/historical-detectors/${detectorId}/edit`);
+  };
+
+  const onStartDetector = async () => {
+    try {
+      // Check to make sure we can start. Current historical detector limit: 2 running per node
+      const clusterStatsResponse = await dispatch(getClusterStats());
+      const nodeCount = get(
+        clusterStatsResponse,
+        'response.nodes.count.data',
+        1
+      );
+      const historicalDetectorsResponse = await dispatch(
+        getHistoricalDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS)
+      );
+      const runningCount = getRunningHistoricalDetectorCount(
+        get(historicalDetectorsResponse, 'response.detectorList')
+      );
+
+      if (nodeCount * 2 <= runningCount) {
+        throw 'only 2 historical detectors can be running per data node in the cluster';
+      }
+
+      await dispatch(startDetector(detectorId));
+      fetchDetector();
+      core.notifications.toasts.addSuccess(
+        `Successfully started the historical detector`
+      );
+    } catch (err) {
+      core.notifications.toasts.addDanger(
+        `There was a problem starting the historical detector: ${err}`
+      );
+    }
+  };
+
+  // We query the task state 5s after making the stop detector call. If the task is still running,
+  // then it is assumed there was an error stopping this task / historical detector.
+  const onStopDetector = async (isDelete: boolean, listener?: Listener) => {
+    try {
+      setIsStoppingDetector(true);
+      await dispatch(stopDetector(detectorId));
+      await waitForMs(HISTORICAL_DETECTOR_STOP_THRESHOLD);
+      setIsStoppingDetector(false);
+      const resp = await dispatch(getDetector(detectorId));
+      if (get(resp, 'response.curState') !== DETECTOR_STATE.DISABLED) {
+        throw 'please try again.';
+      }
+      if (!isDelete) {
+        core.notifications.toasts.addSuccess(
+          'Successfully stopped the historical detector'
+        );
+      }
+      if (listener) listener.onSuccess();
+    } catch (err) {
+      core.notifications.toasts.addDanger(
```

Comment:
Should we set is stopping detector as false if any error happen?

Reply:
Good catch, will call `setIsStoppingDetector(false)` in a `finally` block here.